scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/shared/SchedulerConfig.java

```diff
@@ -135,15 +129,53 @@ public String getRestUrl() {
         return properties.get(REST_URL);
     }
 
+    /**
+     * @return the URL from the window location
+     */
+    public String getWindowsLocationUrl() {
+        String urlFromCurrentLocation = com.google.gwt.user.client.Window.Location.getHref();
+        urlFromCurrentLocation = urlFromCurrentLocation.replace(com.google.gwt.user.client.Window.Location.getPath(), "");
+        return urlFromCurrentLocation;
+    }
+
+    /**
+     * @return the REST_PUBLIC_URL if it is set or take it from the window location
+     */
     @Override
-    public String getRestPublicUrlIfDefinedOrOverridden() { return properties.get(REST_PUBLIC_URL);}
+    public String getRestPublicUrlIfDefinedOrOverridden() {
+        String restPublicUrl = properties.get(REST_PUBLIC_URL);
+        if (restPublicUrl == null) {
+            String restUrlFromCurrentLocation = getWindowsLocationUrl();
+            restUrlFromCurrentLocation += "/rest";
+            return restUrlFromCurrentLocation;
+        }
+        return restPublicUrl;
+    }
 
+    /**
+     * @return the NOVNC_URL if it is set or take it from the window location
+     */
     public String getNoVncUrl() {
-        return properties.get(NOVNC_URL);
+        String noVncUrl = properties.get(NOVNC_URL);
+        if (noVncUrl == null) {
+            String noVncUrlFromCurrentLocation = getWindowsLocationUrl();
```

Comment:
I think more work needs to be done on this config, as the portal port (8080) is not the correct one => 5900